### DISCUSSION TITLE
issue: useFieldsArray witn antd

### DISCUSSION THREAD

**Discussion by babur001** [react-hook-form/react-hook-form#10670] [Q&A]
### Version Number

^7.29.0

### Codesandbox/Expo snack

-

### Steps to reproduce

-

### Expected behaviour

Using useFieldArray with antd Table:

Having somethin like this in my code: 

```tsx
<Table
  rowKey={(row) => row.id}
  columns={columns}
  dataSource={fields.map((field, idx) => ({ id: field.id }))}
  pagination={false}
/>
```

And whenever i append some data it is been added to table row. Then the `field.id` is been consumed as key in all name, as an example `name={`restore.${fieldId}.mxik`}`. But it gives some strange issue in validation where each time i add row errors are swapped and focus is given to another field. But whenever i do  `dataSource={fields.map((field, idx) => ({ id: idx }))}` (swap `field.id` to `idx`) it works.

Is there any unexpected key mismatch i did or is it a bug

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@SaburovBabur 

Please provide a codesadbox url to re-produce the issue.
